backend/src/permissions/permissions.service.spec.ts

```diff
@@ -171,6 +181,24 @@ describe('PermissionsService', () => {
     return note;
   }
 
+  function mockNoteMethods(
+    methodName: MethodType<typeof noteRepo>,
+    implementation: ImplementationType,
+  ) {
+    return jest
+      .spyOn(noteRepo, methodName)
+      .mockImplementationOnce(implementation);
+  }
+
+  function mockEventEmitterMethods(
+    methodName: MethodType<typeof eventEmitter>,
+    implementation: ImplementationType,
+  ) {
+    return jest
+      .spyOn(eventEmitter, methodName)
+      .mockImplementationOnce(implementation);
+  }
+
```

Comment:
Almost, but not exactly. You use mockNoteMethods only for 'save' and mockEventEmitterMethods only for 'emit'.
I don't see a reason for a method-level abstraction if it's not used for other methods.

Reply:
@mrdrogdrog  removed different method level abstraction. Now I'm using a single abstraction for both methods.

Reply:
Sry but i still don't get why you create this abstraction at all. Everywhere where you use these two methods they're used in the exact same way. You always use the mockNoteMethods for mocking the save function and mockEventEmitterMethods to mock the emit function so it expects a certain call.
That's an indication that the parameters are not needed. 
It would be much easier if you create just two functions. One that mocks the save function of the repository and one that mocks the event emitter to expect a certain event. There is no need for more complex abstraction.

You could even move the mock of the save function into the beforeAll because it is done in all tests that need it.

Reply:
@mrdrogdrog, I have removed the abstraction. I only have one mock function for event emitter and one for note repository. 
Please let me know if i have missed anything